src/Main/EvolveLevel.cpp

```diff
@@ -691,6 +691,20 @@ void EvolveLevel( const int lv, const double dTime_FaLv )
 //       12-1. use the average data on fine grids to correct the coarse-grid data
          if ( OPT__FIXUP_RESTRICT )
          {
+#           ifdef DEDT_NU
+
+            TIMING_FUNC(   Flu_FixUp_Restrict( lv, amr->FluSg[lv+1], amr->FluSg[lv], amr->MagSg[lv+1], amr->MagSg[lv],
+                                               NULL_INT, NULL_INT, _TOTAL - _DEDT_NU, _MAG ),
+                           Timer_FixUp[lv],   TIMER_ON   );
+
+#           ifdef LOAD_BALANCE
+            TIMING_FUNC(   LB_GetBufferData( lv, amr->FluSg[lv], amr->MagSg[lv], NULL_INT, DATA_RESTRICT,
+                                             _TOTAL - _DEDT_NU, _MAG, NULL_INT ),
+                           Timer_GetBuf[lv][7],   TIMER_ON   );
+#           endif
+
+#           else
+
```

Comment:
To reduce redundant codes and highlight the difference between the cases w/ and w/o `DEDT_NU`, how about simply defining `ResVar` as follows and then using this variable in both `Flu_FixUp_Restrict()` and `LB_GetBufferData()`?
```
#  ifdef DEDT_NU
   const long ResVar = _TOTAL - _DEDT_NU;
#  else
   const long ResVar = _TOTAL;
#  endif
```

Reply:
Agreed. I use this verbose code to minimize the conflict when merging the master branch. However, it seems that will induce addition bugs such as the correction in `Flu_FixUp_Restrict()` [here](https://github.com/hyschive/gamer-fork/blob/master/src/Fluid/Flu_FixUp_Restrict.cpp#L303-L305) will not be applied.

Now I declare `ResVar` in both `Flu_FixUp_Restrict()` and `LB_GetBufferData()`, and replace `TVarCC` with `ResVar` in

~~~ c
   for (int v=0; v<NCOMP_TOTAL; v++)
      if ( TVarCC & (1L<<v) )    TFluVarIdxList[ NFluVar ++ ] = v;
~~~

Reply:
@hfhsieh That is actually not what I meant. I suggest declaring `ResVar` inside `if ( OPT__FIXUP_RESTRICT ) { ... }` in `EvolveLevel()` and pass it to both `Flu_FixUp_Restrict()` and `LB_GetBufferData()`. Sorry for the confusion!

Reply:
@hyschive I see. However, IIUC, the condition [here](https://github.com/hyschive/gamer-fork/blob/ccsn/src/Fluid/Flu_FixUp_Restrict.cpp#L303-L305) will be false when `dEdt_NU` is defined, and the following check/update will not be executed which include the update of initial temperature guess [here](https://github.com/hyschive/gamer-fork/blob/ccsn/src/Fluid/Flu_FixUp_Restrict.cpp#L348-L364).

Reply:
@hfhsieh I see. Then perhaps we still need to revise `Flu_FixUp_Restrict()` to exclude `_DEDT_NU` from `_TOTAL` [here](https://github.com/hyschive/gamer-fork/blob/ccsn/src/Fluid/Flu_FixUp_Restrict.cpp#L303-L305)? A bit ugly though. What do you think?

Reply:
@hyschive Agree, and I think it is good enough. In the future when the supplement fields are supported, we can remove it then.

Reply:
Updated.